File: backend/Models/AuditLog.cs
namespace InternshipManagement.Api.Models;

public class AuditLog
{
    public Guid Id { get; set; }

    public Guid? ActorUserId { get; set; }

    public string Action { get; set; } = string.Empty;

    public string EntityType { get; set; } = string.Empty;

    public Guid EntityId { get; set; }

    public string? OldStatus { get; set; }

    public string? NewStatus { get; set; }

    public string? Reason { get; set; }

    public string CorrelationId { get; set; } = Guid.NewGuid().ToString();

    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

    public User? ActorUser { get; set; }
}

File: frontend/src/types/auditLog.ts
export interface AuditLogResponse {
    id: string;
  
    actorUserId: string | null;
    actorName: string | null;
    actorEmail: string | null;
  
    action: string;
  
    entityType: string;
    entityId: string;
  
    oldStatus: string | null;
    newStatus: string | null;
  
    reason: string | null;
  
    correlationId: string;
  
    createdAt: string;
  }